openwisp_monitoring/monitoring/base/models.py

```diff
@@ -155,6 +155,8 @@ def check_threshold(self, value, time=None, retention_policy=None, send_alert=Tr
             alert_settings = self.alertsettings
         except ObjectDoesNotExist:
             return
+        if isinstance(time, str):
+            time = utc.localize(datetime.strptime(time, '%Y-%m-%dT%H:%M:%S.%fZ'))
```

Comment:
the only potentially dangerous line I see is this one.

Reply:
To be honest, I agree with you. I feel slightly uncomfortable with timezones and their implications (as to where UTC is to be used and where the local timezone) right now. What I am sure of is that this datetime object needed to be time aware, `Z` implied UTC timezone hence I went with it. Please let me know if I am misunderstanding this anywhere or if you meant dangerous in a different sense, thanks.